spring-integration-core/src/main/java/org/springframework/integration/support/management/graph/IntegrationGraphServer.java

```diff
@@ -79,6 +81,17 @@ public void onApplicationEvent(ContextRefreshedEvent event) {
 	}
 
 	private synchronized Graph buildGraph() {
+		String implementationVersion = IntegrationGraphServer.class.getPackage().getImplementationVersion();
+		if (implementationVersion == null) {
+			implementationVersion = "unknown - is Spring Integration running from the distribution jar?";
+		}
+		Map<String, Object> descriptor = new HashMap<String, Object>();
+		descriptor.put("provider", "spring-integration");
+		descriptor.put("providerVersion", implementationVersion);
+		descriptor.put("providerFormatVersion", GRAPH_VERSION);
+		if (this.applicationContext.containsBean("applicationNameAndVersion")) {
```

Comment:
Both cases can simply be managed by two setters on the `IntegrationGraphServer`.
What is your point?